### PR TITLE
Remove the cycle from babel-types/babel-traverse.

### PR DESCRIPTION
**Pull request by loganfsmyth**

### CODE REVIEW

File: packages/babel-types/src/index.js [hunk 1]
```diff
@@ -431,7 +424,6 @@ export function inherits(child: Object, parent: Object): Object {
   }
 
   t.inheritsComments(child, parent);
-  traverse.copyCache(parent, child);
```
Comment (loganfsmyth) [inline packages/babel-types/src/index.js]: This is the one logic change, but removing it causes no test failures, and nothing obvious comes to mind that would be broken by this.

Comment (danez) [inline packages/babel-types/src/index.js]: The cache is a [WeakMap](https://github.com/babel/babel/blob/dc1f40540de11045bb4714350350e012fee4f93e/packages/babel-traverse/src/cache.js#L1), so I would assume that all logic needs to work even if the cache entry is not there.
👍

Comment (loganfsmyth) [inline packages/babel-types/src/index.js]: Agreed, the only issue would be if there's something in the cache that's critical for the functioning of some transform. I don't think there _should_ be, but that is hard to verify properly.

**Comment by hzoo** [comment]
Sorry! 👍

**Pull request merged**